sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java

```diff
@@ -0,0 +1,1118 @@
+            WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedIdentityBasedAccessToCurrentResourceGroup(
+                String roleDefinitionId);
+        }
+
+        /**
+         * The stage of the container instance definition allowing to specify user assigned managed service identity.
+         */
+        interface WithUserAssignedManagedServiceIdentity {
+            /**
+             * Specifies the definition of a not-yet-created user assigned identity to be associated with the virtual
+             * machine.
+             *
+             * @param creatableIdentity a creatable identity definition
+             * @return the next stage of the definition
+             */
+            WithCreate withNewUserAssignedManagedServiceIdentity(Creatable<Identity> creatableIdentity);
+
+            /**
+             * Specifies an existing user assigned identity to be associate with the container group.
+             *
+             * @param identity the identity
+             * @return the next stage of the definition
+             */
+            WithCreate withExistingUserAssignedManagedServiceIdentity(Identity identity);
+        }
+
+        /** The stage of the container group definition allowing to specify the container group restart policy. */
+        interface WithRestartPolicy {
+            /**
+             * Specifies the restart policy for all the container instances within the container group.
+             *
+             * @param restartPolicy the restart policy for all the container instances within the container group
+             * @return the next stage of the definition
+             */
+            WithCreate withRestartPolicy(ContainerGroupRestartPolicy restartPolicy);
+        }
+
+        /** The stage of the container group definition allowing to specify the DNS prefix label. */
+        interface WithDnsPrefix {
+            /**
+             * Specifies the DNS prefix to be used to create the FQDN for the container group.
+             *
+             * @param dnsPrefix the DNS prefix to be used to create the FQDN for the container group
+             * @return the next stage of the definition
+             */
+            WithCreate withDnsPrefix(String dnsPrefix);
+        }
+
+        /** The stage of the container group definition allowing to specify the network profile id. */
+        interface WithNetworkProfile {
+            /**
+             * Specifies the network profile information for a container group.
+             *
+             * @param subscriptionId the ID of the subscription of the network profile
+             * @param resourceGroupName the name of the resource group of the network profile
+             * @param networkProfileName the name of the network profile
+             * @return the next stage of the definition
+             */
+            DnsConfigFork withNetworkProfileId(
+                String subscriptionId, String resourceGroupName, String networkProfileName);
```

Comment:
Should we provide a method with full id (if network profile got a full id)? And should we call it `withExistingNetworkProfile`?

Reply:
This is a old function, I didn't change it.

Reply:
If we want to do some breaking change, here is the best time. Anyway, up to you.